packages/yoroi-extension/app/components/wallet/send/WalletSendForm.js

```diff
@@ -484,23 +484,19 @@ export default class WalletSendForm extends Component<Props> {
             options={sendAmountOptions}
             {...form.$('selectedAmount').bind()}
             onChange={value => {
-              this.form.$('selectedAmount').value = value
-              if(value === CUSTOM_AMOUNT && this.props.shouldSendAll) {
-                this.props.toggleSendAll();
+              if (value === CUSTOM_AMOUNT) {
+                this.props.updateSendAllStatus(false);
+                this.form.$('amount').clear();
               } else if (value !== CUSTOM_AMOUNT) {
-                this.props.toggleSendAll()
-              }
-              if (this.props.shouldSendAll) {
-                // if we toggle shouldSendAll from true -> false
-                // we need to re-enable the field
-                // and set it to whatever value was used for the sendAll value
+                this.props.updateSendAllStatus(true);
                 this.props.updateAmount(new BigNumber(
```

Comment:
@AhmedIbrahim336 , you have changed the logic so that when "custom amount" gets selected back - the amount pre-calculate for send-all is erased, this is why you removed the comment about preserving that value, but the function call to `.updateAmount` is not removed, even tho it lost it's sense.

BUT! The correct change should be to NOT erase the amount field, the logic was correct before that. Please return the comment back and handle this case correctly:

1. User has "custom amount" set by default
2. User changes to "send all"
3. User input is replaced with the new amount automatically
4. User changes back to "custom amount"
5. The automatic value is preserved and just made editable